geode-core/src/main/java/org/apache/geode/internal/cache/control/InternalResourceManager.java

```diff
@@ -637,7 +638,9 @@ private ThreadsMonitoring getThreadMonitorObj() {
    */
   public void addStartupTask(CompletableFuture<Void> startupTask) {
     Objects.requireNonNull(startupTask);
-    startupTasks.add(startupTask);
+    synchronized (startupTasks) {
```

Comment:
Can this be achieved by changing startupTasks to CopyOnWriteArrayList.

Reply:
`CopyOnWriteArrayList` makes a fresh copy of the underlying array for every mutative operation such as `add`. This is too costly for this use case.